modules/secure_storage_connector/README.md

```diff
@@ -0,0 +1,71 @@
+# Weights & Biases Secure Storage Connector Module
+
+This is a Terraform module for provisioning a team-level azure storage container to be used with Weights and Biases.
+A resource group and managed identity used to auth to the container will also be created by default. To use this container with Weights and Biases
+dedicated azure deployment, you need to specify the variable for the OIDC issuer URL from the AKS cluster that will connect to this container. 
+
+## Azure Services Used
+
+- Microsoft Entra ID (formerly Azure Active Directory)
+- Azure Blob Storage
+
+## How to Use This Module
+
+- Create a Terraform configuration that pulls in this module and specifies
+  values of the required variables:
+
+```
+provider "azurerm" {
+  features {} 
+  subscription_id = "<your subscription id>"
+  tenant_id       = "<your tenant id>"
+}
+
+module "secure_storage_connector" {
+  source          = "wandb/wandb/azure/modules/secure_storage_connector"
+  namespace       = "<prefix for naming Azure resources>"
+  location        = "<Azure Region where resources will be created>"
+  oidc_issuer_url = "<OIDC issuer URL from the AKS cluster that is meant to connect to this container. Make sure to include the trailing '/'>"
+}
+```
+
+- Run `terraform init`
+- Run `terraform apply -target=azurerm_user_assigned_identity.default`
+  - The managed identity needs to be provisioned before it can be passed into a module that this module references. 
```

Comment:
otherwise you'll get this error
```
│ Error: Invalid count argument
│
│   on ../storage/main.tf line 49, in resource "azurerm_role_assignment" "account":
│   49:   count = var.managed_identity_principal_id != "" ? 1 : 0
│
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first
│ apply only the resources that the count depends on.
```

Reply:
This seems weird because I think identity stuff in other parts of code and its not have this issue? Maybe I don't have full context but we really should have the system apply in 1 go

Reply:
thanks for calling out. fixed